docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -47,53 +47,35 @@ aws s3 cp <DIRECTORY> s3://<TENANT>/<DIRECTORY>
+If you enable the sharding feature of Mimir's split-and-merge compactor, compactor adds a special label to identify the compactor-shard.
 
-- If you are using Docker, run the following command:
+In the Grafana Mimir 2.2 (or later) release, blocks no longer have a label that identifies the tenant. The compactor-shard label is still used, if sharding of blocks during compaction is enabled.
 
-```bash
-docker pull grafana/metaconvert:latest
-```
-
-- If you are using a release binary, download the appropriate release asset for your operating system and architecture and make it executable. For Linux with the AMD64 architecture, run the following command:
-
-```bash
-curl -LO https://github.com/grafana/mimir/releases/latest/download/metaconvert-linux-amd64
-chmod +x metaconvert-linux-amd64
-```
-
-- If you are using Go, run the following command:
-
-```bash
-go install github.com/grafana/mimir/cmd/metaconvert@latest
-```
-
-### Running `metaconvert`
-
-> **Warning:** The `metaconvert` tool modifies objects in place.
-> Ensure you enable bucket versioning or have backups before running the tool.
-
-To run `metaconvert`, provide it with the bucket configuration. Use `metaconvert -h` to list available parameters.
+> **Note**: Blocks from Prometheus do not have any external labels stored in them; only blocks from Thanos use labels.
 
-1. Complete one of the following steps to run `metaconvert` in a dry-run mode, which lists blocks for migration.
+## Considerations on Thanos specific features
 
-   - If you are using Docker, run the following command:
+> **Note**: Thanos requires that Prometheus is configured with external labels.
+> When the Thanos sidecar uploads blocks, it includes the external labels from Prometheus in the `meta.json` file inside the block.
+> When you query the block, Thanos injects Prometheus’ external labels in the series returned in the query result. Thanos also uses labels for the deduplication of replicated data.
 
-   ```bash
-   docker run grafana/metaconvert -backend=gcs -gcs.bucket-name=bucket -tenant=anonymous -dry-run
-   ```
+If you want to use existing blocks from Thanos by Grafana Mimir, there are some considerations:
 
-   - If you are using a local binary, run the following command:
+**Grafana Mimir doesn't inject external labels into query results.**
+This means that blocks that were originally created by Thanos will not include their external labels in the results when queried.
+If you need to have external labels in your query results, this is currently not possible to achieve in Grafana Mimir.
 
-   ```bash
-   ./metaconvert -backend=filesystem -filesystem.dir=/bucket -tenant=anonymous -dry-run
-   ```
+**Grafana Mimir will not respect deduplication labels configured in Thanos when querying the blocks.**
+For the best results please only upload Thanos blocks from single Prometheus replica from each HA pair.
```

Comment:
What does "best results" mean here? If I upload all Thanos blocks, are query results incorrect or just less performant? I think they're just less performant, so I would clarify it ("For the best query performance...").

Reply:
Good point.